pulsar-client-api/src/main/java/org/apache/pulsar/client/api/Message.java

```diff
@@ -245,4 +245,20 @@
      * @since 2.8.0
      */
     void release();
+
+    /**
+     * Get broker publish time from broker entry metadata.
```

Comment:
Please explain that this value is not available if the broker is older or it does not enable this feature.

What about Optional<Long> ?

Reply:
Thanks. I've changed the return value from long to Optional<Long> and improve the doc.